File: backend/mvc/Controllers/HomeController.cs
//HO TENUTO IL FILE SOLO PER CONFRONTARLO CON L'IMPLEMENTAZIONE IN NODE VEDERE HomeController.js

using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;
using System.Web.Mvc;

namespace MVCSample.Controllers
{
    public class HomeController : Controller
    {
        public ActionResult Index() {
            return View();
        }
    }
}

File: backend/mvc/Controllers/HomeController.js
// Inizializzo un oggetto per configurare il routing i middleware usati in questo "Controller" MVC/Home
var HomeController = require('express').Router();

HomeController.get('/Index', function(req,res) {		
	// uso l'helper res.View() esposto dal middleware per renderizzare con Razor il file cshtml	
	res.View('Home/Index.cshtml');
});

HomeController.get('/About', function(req,res) {
	// uso l'helper res.View() esposto dal middleware per renderizzare con Razor il file cshtml	+ passandogli il @model	
	res.View('Home/About', require('../Models/HostInfo'));
});

exports = module.exports = HomeController;